imports/plugins/core/catalog/server/methods/catalog.js

```diff
@@ -937,6 +937,8 @@ Meteor.methods({
       return numFlaggedAsDeleted;
     }
 
+    Hooks.Events.run("afterRemoveCatalogProduct", this.userId, product);
+
```

Comment:
It looks like this would have been already done during `productsWithVariants.forEach` above?

Reply:
@aldeed the loop flags the product as deleted in the `Products` collection, the hook does the same in the `Catalog` collection.

Reply:
@willopez already said it but yes, it does run in the variants, but it wasn't running for the product itself, and since we only push the parents to the `Catalog`, we need to run it for the overall product as well.